acct/setup-otp

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+if [ "$#" -ne 1 ]; then
+    echo "Usage: setup-otp <username>"
+    exit 1
+fi
+
+username="$1"
+
+# get last ocf-tagged public ID
+IDHex=$(sudo cut -d: -f2- /etc/otp/yubikeys | tr : \\n | grep vvfuckcrv | tr "cbdefghijklnrtuv" "0123456789ABCDEF" | sort | tail -1)
+
+while true; do
+    # increment to get next ID
+    IDHex=$(echo "obase=ibase=16;${IDHex}+1" | bc)
```

Comment:
Here's to hoping we never run into a race condition with this, lol

Reply:
The utility will fall hard if the public ID is taken, so if that happens you just rerun the script. The only thing is that you'll have to use slot 1 instead of "fresh".
However, a race on this line is not a problem because if I key is registered but not yet present in the file it will get skipped over by the curl incantation, and we'll increment again.